backend/dataall/core/environment/cdk/environment_stack.py

```diff
@@ -165,30 +165,44 @@ def __init__(self, scope, id, target_uri: str = None, **kwargs):
                 f'arn:aws:iam::{self._environment.AwsAccountId}:role/{self.pivot_role_name}',
             )
 
-        # Environment S3 Bucket
-        default_environment_bucket = s3.Bucket(
+        # Environment Logging S3 Bucket
+        default_environment_log_bucket = s3.Bucket(
             self,
-            'EnvironmentDefaultBucket',
-            bucket_name=self._environment.EnvironmentDefaultBucketName,
+            'EnvironmentDefaultLogBucket',
+            bucket_name=self._environment.EnvironmentLogsBucketName,
             encryption=s3.BucketEncryption.S3_MANAGED,
             removal_policy=RemovalPolicy.RETAIN,
             block_public_access=s3.BlockPublicAccess.BLOCK_ALL,
             versioned=True,
             enforce_ssl=True,
         )
-        default_environment_bucket.policy.apply_removal_policy(RemovalPolicy.RETAIN)
-        self.default_environment_bucket = default_environment_bucket
-
-        default_environment_bucket.add_to_resource_policy(
+        default_environment_log_bucket.policy.apply_removal_policy(RemovalPolicy.RETAIN)
```

Comment:
Isn't this already define in line 174?

Reply:
removal policy applies to the bucket only - the assocaited policy would get deleted unless we include this line 
```
        default_environment_log_bucket.policy.apply_removal_policy(RemovalPolicy.RETAIN)
```

which then potentially exposes our bucket